CRM/Core/Reference/Dynamic.php

```diff
@@ -24,7 +24,10 @@ public function matchesTargetTable($tableName) {
   public function getTargetEntities(): array {
     $targetEntities = [];
     $bao = CRM_Core_DAO_AllCoreTables::getClassForTable($this->refTable);
-    $targetTables = (array) $bao::buildOptions($this->refTypeColumn);
+    $targetTables = $this->getTargetTable() ?: (array) $bao::buildOptions($this->refTypeColumn);
+    if (strstr($targetTables, '::')) {
```

Comment:
This is causing a test failure in `api\v4\Action\FkJoinTest`, because `$targetTables` might be an array.
```suggestion
    if (is_string($targetTables) && strstr($targetTables, '::')) {
```

Reply:
Incidently, `api\v4\Action\FkJoinTest` would be a good spot to add a unit test for this PR.